### ISSUE TITLE
Add cache shards to increase performance

### ISSUE THREAD

**Issue by Planxnx** [jellydator/ttlcache#93]
add cache shards will reduce r/w blocking by mutex.
Example: https://github.com/orcaman/concurrent-map/blob/master/concurrent_map.go

**Comment by sammy-hughes** [comment]
Given the internalization of a mutex, this would seem to be a valuable feature. Is this feature being considered?

**Comment by swithek** [comment]
Yes, though there are no plans for its release yet.

**Comment by sammy-hughes** [comment]
Yep. The distance between "that's a good idea" and "all done!" can be quite the doozy. Thanks for confirming that this is somewhere in the roadmap.

**Comment by Planxnx** [comment]
I'm still waiting 👀

**Comment by jfberry** [comment]
We made a generic wrapper for ttlcache  (not all functions, but the ones we needed) to shard it. It increased performance massively.

https://github.com/UnownHash/Golbat/blob/main/decoder/sharded_cache.go
